fs2loader/src/main/scala/com.snowplowanalytics.snowplow.storage.bigquery.fs2loader/Fs2Loader.scala

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2018-2020 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.storage.bigquery.fs2loader
+
+import cats.syntax.all._
+import cats.effect.{Concurrent, ContextShift, ExitCode, IO, Timer}
+
+import fs2.{Pipe, Stream}
+import fs2.concurrent.Queue
+
+import com.snowplowanalytics.snowplow.badrows.BadRow
+import com.snowplowanalytics.snowplow.storage.bigquery.common.Codecs.toPayload
+import com.snowplowanalytics.snowplow.storage.bigquery.common.Config.Environment
+import com.snowplowanalytics.snowplow.storage.bigquery.fs2loader.events._
+import com.snowplowanalytics.snowplow.storage.bigquery.fs2loader.sinks.PubSub
+import com.snowplowanalytics.snowplow.storage.bigquery.fs2loader.sinks.PubSub.{WriteBadRow, WriteObservedTypes}
+import com.snowplowanalytics.snowplow.storage.bigquery.loader.LoaderRow
+
+import org.slf4j.LoggerFactory
+
+import com.snowplowanalytics.snowplow.analytics.scalasdk.Data.ShreddedType
+
+import scala.concurrent.duration._
+
+object Fs2Loader {
+  private val MaxConcurrency = 5
+  private val GroupByN       = 5
+  private val TimeWindow     = 2.seconds
+
+  /**
+    * Aggregate observed types when a specific number is reached or time passes, whichever happens first.
+    * @param groupByN Number of elements that trigger aggregation.
+    * @param timeWindow Time window to aggregate over if n limit is not reached.
+    * @tparam ST In a prod setting, this should be [[ShreddedType]].
+    * @return A pipe that does not change the type of elements but discards non-unique values within the group.
+    */
+  def aggregateTypes[ST](
+    groupByN: Int,
```

Comment:
If we're passing in the constant then we can get rid of the argument.

Also likely we'll want to stick with high enough `groupByN` contstant and some sensible `TimeWindow`, like 1-2 mins (primary loader has 5 mins, but emits much faster for some reasons).

Reply:
Ah, so you left them as arguments because you want to use them in tests. Makes sense.